### PR TITLE
COS-2613: kola: fix rhaos-pkgs-match-openshift

### PR DESCRIPTION
**Pull request by c4rt0**
See: https://github.com/openshift/os/issues/1427

### CODE REVIEW

File: tests/kola/version/rhaos-pkgs-match-openshift
Comment (jlebon) [inline tests/kola/version/rhaos-pkgs-match-openshift:12]: The way this conditional is structured makes it easy to have false positives. E.g. if `rpm` is completely broken for whatever reason, the test will still succeed.

I would reflow it as e.g.:

```bash
rpm -qa | grep rhaos > rhaos-packages.txt
if grep -qv ... rhaos-packages.txt; then
  fatal ...
fi
```

File: tests/kola/version/rhaos-pkgs-match-openshift
Comment (c4rt0) [reply tests/kola/version/rhaos-pkgs-match-openshift]: Ah, I didn't think about that. I assumed OPENSHIFT_VERSION was somehow available already. Thanks @mike-nguyen 🙏🏻

File: tests/kola/version/rhaos-pkgs-match-openshift
Comment (c4rt0) [reply tests/kola/version/rhaos-pkgs-match-openshift:15]: Thanks @jlebon - if the considered rpm is completely broken, is it not the case that it will still show up in the `rhaos-packages.txt` ?

Comment (c4rt0) [reply tests/kola/version/rhaos-pkgs-match-openshift]: Regardless, I updated the code as per the suggestion.

File: tests/kola/version/rhaos-pkgs-match-openshift
Comment (cverna) [inline tests/kola/version/rhaos-pkgs-match-openshift:17]: I think matching rhaos${OPENSHIFT_VERSION} will be more solid. That will avoid to match the OCP version in the package version. 
For example if OCP version is 4.16 and a package has a version containing 4.16 this would not trigger an error `foobar-4.16.4-0.1.rhaos4.13.el9.x86_64`

```suggestion
if grep -qv "rhaos${OPENSHIFT_VERSION}" rhaos-packages.txt; then
```

Comment (cverna) [reply tests/kola/version/rhaos-pkgs-match-openshift:17]: Also I think dropping the -q (quiet) option might be good so that when the test fail we can see which packages trigger the error in the logs. That makes it easier to investigate.

```suggestion
if grep -v "${OPENSHIFT_VERSION}" rhaos-packages.txt; then
```

Comment (c4rt0) [reply tests/kola/version/rhaos-pkgs-match-openshift]: Thank you - I agree with the above.

**Comment by jlebon** [comment]
CI failure looks relevant. I'll let you look at the logs and do the appropriate fix. :)

**Comment by c4rt0** [comment]
Test now fail's as expected:

```
Mar 12 14:07:39 qemu0 kola-runext-rhaos-pkgs-match-openshift[20797]: + grep -v rhaos4.16 /tmp/rhaos-packages.txt
Mar 12 14:07:39 qemu0 kola-runext-rhaos-pkgs-match-openshift[20801]: conmon-2.1.7-3.2.rhaos4.14.el9.x86_64
Mar 12 14:07:39 qemu0 kola-runext-rhaos-pkgs-match-openshift[20801]: containers-common-1-37.rhaos4.13.el9.x86_64
Mar 12 14:07:39 qemu0 kola-runext-rhaos-pkgs-match-openshift[20801]: conmon-rs-0.5.1-5.rhaos4.13.git.el9.x86_64
Mar 12 14:07:39 qemu0 kola-runext-rhaos-pkgs-match-openshift[20797]: + fatal 'Error: rhaos packages do not match OpenShift version'
Mar 12 14:07:39 qemu0 kola-runext-rhaos-pkgs-match-openshift[20797]: + echo 'Error: rhaos packages do not match OpenShift version'
Mar 12 14:07:39 qemu0 kola-runext-rhaos-pkgs-match-openshift[20797]: Error: rhaos packages do not match OpenShift version
```

File: tests/kola/version/rhaos-pkgs-match-openshift
Comment (travier) [inline tests/kola/version/rhaos-pkgs-match-openshift:19]: Nit: Missing an empty line at the end of the file

**Comment by travier** [comment]
We'll also likely have to backport this fix down to 4.14 at least

File: tests/kola/version/rhaos-pkgs-match-openshift
Comment (c4rt0) [reply tests/kola/version/rhaos-pkgs-match-openshift]: Correctred

**Comment by c4rt0** [comment]
> OK, so we'll have to ask Jindrich from the node/container tools team to update those packages for us.

I'm not sure which Jindrich exactly to ask about this. Could you please point me to the right person or reach out to him yourself?

> We'll also likely have to backport this fix down to 4.14 at least, maybe even down to 4.12

I did backporting a few months ago - will dig into documentation on how to do it again.

**Review by jbtrystram — APPROVED** [review]
LGTM

**Comment by c4rt0** [comment]
After a fresh build I see that one of the packages is still outdated, hence this test fails:

```
[core@cosa-devsh ~]$ date
Fri Mar 29 18:35:35 UTC 2024
[core@cosa-devsh ~]$ rpm -qa | grep conmon-rs
conmon-rs-0.5.1-6.rhaos4.13.el9.x86_64
```

Requested one more update from Jindrich.

**Comment by travier** [comment]
```
Apr 08 14:10:08 qemu0 kola-runext-rhaos-pkgs-match-openshift[8099]: + grep -v rhaos4.16 /tmp/rhaos-packages.txt
Apr 08 14:10:08 qemu0 kola-runext-rhaos-pkgs-match-openshift[8103]: conmon-rs-0.5.1-6.rhaos4.13.el9.x86_64
Apr 08 14:10:08 qemu0 kola-runext-rhaos-pkgs-match-openshift[8099]: + fatal 'Error: rhaos packages do not match OpenShift version'
```

**Comment by c4rt0** [comment]
The latest build of the [updated](https://brewweb.engineering.redhat.com/brew/taskinfo?taskID=60877595) `conmon-rs` succeeded. Retesting to verify if tests pass.
/retest

**Comment by c4rt0** [comment]
🎉

File: tests/kola/version/rhaos-pkgs-match-openshift
Comment (dustymabe) [inline tests/kola/version/rhaos-pkgs-match-openshift:17]: not much need for the intermediate file IIUC:

```suggestion
rpm -qa | grep "rhaos" | grep -v "rhaos${OPENSHIFT_VERSION}"; then
```

Comment (dustymabe) [reply tests/kola/version/rhaos-pkgs-match-openshift:17]: can do that in a followup if you like.

**Comment by c4rt0** [comment]
Awesome, will do! Thanks @dustymabe.

**Comment by travier** [comment]
Thanks for persisting here Adam. Let's get this backported to where it's needed (I don't remember which versions need it).

/cherry-pick release-4.15

**Comment by c4rt0** [comment]
> Thanks for persisting here Adam. Let's get this backported to where it's needed (I don't remember which versions need it).
> 
> /cherry-pick release-4.15

I ran a fresh build of 4.15 today, unfortunately, `conmon-rs` isn't updated for this version yet. I will ask the maintainers if it's possible to update this one too. 
For now:
```
Last login: Fri May  3 12:14:39 2024
[core@cosa-devsh ~]$ rpm -qa | grep conmon-rs
conmon-rs-0.5.1-6.rhaos4.13.el9.x86_64
[core@cosa-devsh ~]$ cat /etc/os
os-release  ostree/     
[core@cosa-devsh ~]$ cat /etc/os-release 
NAME="Red Hat Enterprise Linux CoreOS"
ID="rhcos"
ID_LIKE="rhel fedora"
VERSION="415.92.202405031209-0"
VERSION_ID="4.15"
VARIANT="CoreOS"
VARIANT_ID=coreos
PLATFORM_ID="platform:el9"
PRETTY_NAME="Red Hat Enterprise Linux CoreOS 415.92.202405031209-0 (Plow)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:redhat:enterprise_linux:9::coreos"
HOME_URL="https://www.redhat.com/"
DOCUMENTATION_URL="https://docs.openshift.com/container-platform/4.15/"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="OpenShift Container Platform"
REDHAT_BUGZILLA_PRODUCT_VERSION="4.15"
REDHAT_SUPPORT_PRODUCT="OpenShift Container Platform"
REDHAT_SUPPORT_PRODUCT_VERSION="4.15"
OPENSHIFT_VERSION="4.15"
RHEL_VERSION="9.2"
OSTREE_VERSION="415.92.202405031209-0"
```

**Comment by travier** [comment]
We may also decide to remove it from 4.15 if no one is using it there, but I'm not 100% that's the case.

**Comment by c4rt0** [comment]
`conmon-rs` is now updated (that was QUICK). When running tests however on 4.15 I noticed that `containers-common` will also not allow the tests to pass. Enquiry with maintainers is now updated, let's see how this goes.

```
May 03 12:41:35 qemu0 kola-runext-rhaos-pkgs-match-openshift: + grep -v rhaos4.15 /tmp/rhaos-packages.txt
May 03 12:41:35 qemu0 kola-runext-rhaos-pkgs-match-openshift: containers-common-1-37.rhaos4.13.el9.x86_64
May 03 12:41:35 qemu0 kola-runext-rhaos-pkgs-match-openshift: conmon-rs-0.5.1-6.rhaos4.13.el9.x86_64
```

**Comment by c4rt0** [comment]
> We'll also likely have to backport this fix down to 4.14 at least, maybe even down to 4.12

/cherry-pick release-4.13

**Comment by c4rt0** [comment]
Is there a way to close the automatic [cherry-pick](https://github.com/openshift/os/pull/1512) on behalf of the [manual one](https://github.com/openshift/os/pull/1515)?
cc. @travier

**Pull request merged**
